docs/signing-tool-spec.md

```diff
@@ -0,0 +1,129 @@
+
+While some of this could be done via an MSBuild task, a CLI tool is preferable to MSBuild tasks for a couple reasons:
+
+- **Performance:** During a build, many binary artifacts are created that need to be signed. A multi-targeted NuGet package may contain several `.dll` files. An application will likely contain more than one file that needs to be signed. It's much more efficient to pass them all to a signing tool where parallelism is possible than to sign during the inner-loop.
+- **Security:** Code signing is a sensitive operation that requires credentials/secrets. We do not want secrets to be in the MSBuild pipeline as that makes any logs contain those secrets. Ideally, a CI pipeline should contain a separate stage for code signing to ensure that credentials are never unintentionally exposed to a build stage.
+- **Platform:** Authenticode is currently limited to Windows. Thus, while it's possible to sign a NuGet or VSIX cross-platform, the DLL's inside can't be signed unless running on Windows. With the NuGet packages being developer-only artifacts--they're not shipped with the apps--it's critical that the DLL's inside are also signed. Builds for binaries may run on any platform, but as signing is a discreet step in most CI pipelines, it's possible to use a Windows agent for this task.
+
+### Roadmap
+
+The scope of the preview release will be limited to the existing funtionality currently in the service. The remaining functionality in this spec will be delivered for the initial release. The .NET Foundation has a dependency on this tool being delivered by the end of 2022.
+
+#### Preview
+
+**Goals**
+
+- Support for Authenticode, VSIX, NuGet (author signature), ClickOnce
+- Only run on Windows x64.
+- Support a single certificate for all files in the operation.
+
+**Non-Goals**
+
+- Strong Name signing won't be in v1; guidance is to use an snk not based on a cert. If easy, perhaps can revisit.
+- Support for Microsoft internal builds and signing services.
+- Containers, including Notary v2 support.
+- Extensibility. v1 will support different signing providers.
+- Support for platforms other than Windows x64. Future work will be required to support ARM64 and non-Windows hosts. Support for certain file types may be limited due to platform support.
+- Offline distribution.
+
+#### v1
+
+**Goals**
+
+- Extensibility mechanism to support different code signing providers with a dynamic lookup so the core client remains agnostic of the backend
+- Offline distribution for core plus backend provider
+- Three providers: Certificate Store, Azure Key Vault, Azure Code Signing
+- Support for additional formats: [.HLKX](https://github.com/dotnet/sign/issues/422), [VBA](https://github.com/dotnet/sign/issues/364)
+- Verification of signatures
+
+
+### Inputs
```

Comment:
From here to the end doesn't belong in this spec.  It belongs in a separate CLI syntax and options spec.